diff --git a/client/CMakeLists.txt b/client/CMakeLists.txt
index 7392e4c8..f6e48ae5 100644
--- a/client/CMakeLists.txt
+++ b/client/CMakeLists.txt
@@ -241,22 +241,4 @@ endif()
 
 if(UNIX AND (NOT APPLE))
     install(IMPORTED_RUNTIME_ARTIFACTS wsnet DESTINATION lib)
-    # for some reason the installation via the command IMPORTED_RUNTIME_ARTIFACTS does not work for openssl
-    # so for now we just copy the openssl libs
-    install(FILES ${_OPENSSL_LIBDIR}/libcrypto.so.3 DESTINATION lib)
-    install(FILES ${_OPENSSL_LIBDIR}/libssl.so.3 DESTINATION lib)
-    install(FILES ${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/bin/oqsprovider.so DESTINATION lib)
-
-    # copy openvpn
-    install(FILES ${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/tools/openvpn/sbin/openvpn
-            RENAME windscribeopenvpn
-            DESTINATION .
-            PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-    )
-    # copy ctrld
-    install(FILES ${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/tools/ctrld/ctrld
-            RENAME windscribectrld
-            DESTINATION .
-            PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-    )
 endif()
